### DISCUSSION TITLE
How to retrieve dynamically defined atoms from child components using store?

### DISCUSSION THREAD

**Discussion by trishagonzales** [pmndrs/jotai#2133] [Q&A] [answered]
Since dynamic atoms are defined within render function, its name or variable wouldn't be available to use in useStore in other components. How is it usually retrieved and consumed in child components?
```tsx
function Parent(props: PropsWithChildren & AtomState) {
   const { children, ...rest } = props;
   const atomConfig = useMemo(() => atom(rest), []);
   store.set(atomConfig, rest);

   return (
      <Provider store={store}>
         {children}
      </Provider>
   )
}

function Child() {
   const store = useStore();
   const [state, setState] = useAtom(store.get(/* atom variable (config) */));

   return (
      // ...
   )
}
```

**Comment by dai-shi** [comment, marked as answer]
You want to simply use React Context and pass `atomConfig`.
btw, you don't need `useStore` in Children.

**Reply by trishagonzales** [reply]
Thanks for response,
I thought i could replace react context with jotai provider because i can't seem to make typescript generics work with context.

**Reply by dai-shi** [reply]
You could use atoms-in-atom pattern: https://jotai.org/docs/guides/atoms-in-atom
